ddp-workspace/projects/ddp-dsm-ui/src/app/scan/scan.component.ts

```diff
@@ -110,20 +108,22 @@ export class ScanComponent implements OnInit {
 
   private validateValue(labelValue: string, position: number, isLeft: boolean): boolean {
     let isDuplicate = false;
-    for (let i = 0; i < this.scanPairsValue.length; i++) {
-      if (i !== position) {
-        if (this.scanPairsValue[position].leftValue != null && labelValue === this.scanPairsValue[i].leftValue
-          || this.scanPairsValue[position].rightValue != null && labelValue === this.scanPairsValue[i].rightValue) {
-          isDuplicate = true;
-        }
-      } else {
-        if (isLeft) {
-          if (this.scanPairsValue[position].rightValue != null && labelValue === this.scanPairsValue[i].rightValue) {
+    if(labelValue !== '') {
```

Comment:
Just found:
This function is called too many times, which, I think, is not necessary:

https://user-images.githubusercontent.com/77500504/207807710-832df54c-8207-4d95-98bc-e9605e9bde6f.mov

Reply:
I changed it so the function is no longer called when the page is loaded, but the other calls are a result of changing the  condition for adding a new scan pair. Specifically it is the change made to scan-pair-component.ts. I am currently waiting for Simone to reply to your question